ios/browser/api/common/common_operations.mm

```diff
@@ -105,46 +117,43 @@ - (void)loadURLRequest:(const std::string&)url
             return;
           };
           const auto strongSelf = weakSelf;
-
-          const auto response = (NSHTTPURLResponse*)urlResponse;
-          std::string body;
-          if (data && data.length > 0) {
-            body =
-                std::string(static_cast<const char*>(data.bytes), data.length);
-          }
-          std::string errorDescription;
-          if (error) {
-            errorDescription = error.localizedDescription.UTF8String;
-          }
-          // For some reason I couldn't just do `base::flat_map<std::string,
-          // std::string> responseHeaders;` due to base::flat_map's non-const
-          // key insertion
-          auto* responseHeaders =
-              new base::flat_map<std::string, std::string>();
-          [response.allHeaderFields
-              enumerateKeysAndObjectsUsingBlock:^(NSString* _Nonnull key,
-                                                  NSString* _Nonnull obj,
-                                                  BOOL* _Nonnull stop) {
-                if (![key isKindOfClass:NSString.class] ||
-                    ![obj isKindOfClass:NSString.class]) {
-                  return;
+          strongSelf->_taskRunner->PostTask(
```

Comment:
Why is this change required?

Reply:
Because the callbacks into ledger have to be called on the same task runner that ledger runs on